src/coreclr/vm/excep.cpp

```diff
@@ -6557,6 +6559,8 @@ void HandleManagedFaultNew(EXCEPTION_RECORD* pExceptionRecord, CONTEXT* pContext
     args[ARGNUM_0] = DWORD_TO_ARGHOLDER(exceptionCode);
     args[ARGNUM_1] = PTR_TO_ARGHOLDER(&exInfo);
 
+    FirstChanceExceptionNotification();
```

Comment:
The legacy EH rethrows the exception as the original exception code. This will rethrow it as `EXCEPTION_COMPLUS`. 

I look at the EH rethrowing in the legacy EH as a shortcoming, not as a feature. I am not sure whether it makes sense to replicate it here.

Reply:
Let me remove it for this PR as it is unrelated to the actual fix and reason about this later.